org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDiagnosticsHandler.java

```diff
@@ -232,6 +232,24 @@ private static DiagnosticSeverity convertSeverity(IProblem problem) {
 	@SuppressWarnings("restriction")
 	private static Range convertRange(IOpenable openable, IProblem problem) {
 		try {
+			if (problem.getID() == IProblem.UndefinedType && openable instanceof ICompilationUnit) {
+				ICompilationUnit cu = (ICompilationUnit) openable;
+				if (cu.getBuffer() != null) {
+					int start = problem.getSourceStart();
+					while (start > 0) {

```

Comment:
is the `while (start > 0)` needed here ? Could it become an `if` ?

Reply:
Fixed.